doc/sonic-installer/SONiC_Image_Installation_Verification.md

```diff
@@ -0,0 +1,172 @@
+                                  installation
+  --swap-mem-size INTEGER         SWAP memory space size  [default: (1024
+                                  MiB)] NOTE: this argument is mutually
+                                  exclusive with arguments: skip_setup_swap
+  --total-mem-threshold INTEGER   If system total memory is lower than
+                                  threshold, setup SWAP memory  [default:
+                                  (2048 MiB)] NOTE: this argument is mutually
+                                  exclusive with arguments: skip_setup_swap
+  --available-mem-threshold INTEGER
+                                  If system available memory is lower than
+                                  threhold, setup SWAP memory  [default: (1200
+                                  MiB)] NOTE: this argument is mutually
+                                  exclusive with arguments: skip_setup_swap
+  --help                          Show this message and exit.
+
+```
+
+# 4 Design and Flow Diagrams
+
+## 4.1 Flow Diagrams
+The following figure shows how to intall an image through SONiC CLI.
+```
++-----------------+  
+|   SONiC ClI     |            
+|                 <-----------+
+|  Image Install  |           |
++-----------------+           |
+         |                    |
+         V                    |
++-----------------+           |
+|      Image      |---->NO----+
+|                 |             +------------+
+|   Verification  |---->YES-----> Installing |
++-----------------+             +------------+
+```
+
+## 4.2 Design
+
+### 4.2.1 Generate platform list
+We need to generate a file `platforms_asic`. This file contains the list of supported platforms and is generated during the process of image build. The file is kept in built image.
+
+### 4.2.2 Verification
+To decide whether an image is of a correct type of ASIC during installation, the installer will verify if the current platform is inside to be installed image's built-in file `platforms_asic`. If not, verification would fail because it means the current platform is of a different ASIC type from the built image. If yes, it will pass the verification process and do other image checks.
+
+### 4.2.3 Skip platform check
+We also provide a way to bypass the image verification. During sonic-install install process, user can add an 'force install' option to bypass platform check if sonic-installer detects that the image is of a different type of ASIC from the current platform. 
+
+As `--force` option has been used for image secure check, we use  `--skip-platform-check` option to handle our verification.
+
+# 5 Error Handling
+
+# 6 Serviceability and Debug
+
+# 7 Warm Boot Support
+
+# 8 Scalability
+
+# 9 Tests
+
+## 9.1 Unit Tests
+| Test Case | Description |
+| --------- | ----------- |
+| 1         | Install broadcom image on vs platform.|
+| 2         | Install broadcom-dnx image on broadcom platform.|
```

Comment:
>broadcom [](http://example.com/codeflow?start=44&length=8)

How about the reverse case?

Reply:
There are lots of other images such as barefoot, centec etc.
Do we need to add all the ASIC type in the unit test? Or just add one  reverse broadcom and broadcom-dnx is good?

Reply:
Just broadcom image on broadcom-dnx platform